lib/phase/flatten.js

```diff
@@ -30,6 +30,42 @@ function merge(newJob, oldJob) {
     const oldSecrets = oldJob.secrets || [];
 
     newJob.secrets = [...new Set([...newSecrets, ...oldSecrets])];  // remove duplicate
+
+    if (fromTemplate && oldJob.steps) {
+        let stepName;
+        let preStepName;
+        let postStepName;
+        const mergedSteps = [];
+
+        // convert steps from oldJob from array to object for faster lookup
+        const oldSteps = oldJob.steps.reduce((obj, item) => {
+            const key = Object.keys(item)[0];
+
+            obj[key] = item[key];
+
+            return obj;
+        }, {});
+
+        for (let i = 0; i < newJob.steps.length; i += 1) {
+            stepName = Object.keys(newJob.steps[i])[0];
```

Comment:
Same comment as above too. There is only 1 key in this object, and it's the `stepName`.